lib/notifications/fcm/fcm_notifications_listener.dart

```diff
@@ -127,34 +109,20 @@ class FcmNotificationsListener {
     if (!Platform.isAndroid) {
       return;
     }
-
-    String? notificationTitle = message.notification?.title;
-    String? notificationBody = message.notification?.body;
-    const AndroidNotificationDetails androidPlatformChannelSpecifics =
-    AndroidNotificationDetails(
-      CHANNEL_ID,
-      CHANNEL_NAME,
-      channelDescription: CHANNEL_DESCRIPTION,
-      importance: Importance.max,
-      priority: Priority.high,
-    );
-    const NotificationDetails platformChannelSpecifics =
-    NotificationDetails(android: androidPlatformChannelSpecifics);
-    await flNotification.show(
-        0, notificationTitle, notificationBody, platformChannelSpecifics);
+    serviceLocator
```

Comment:
Add it in the constructor. Don't use the service locator inside. If you have a circular dependency mark it as late and use a setter method to initialize it from the service locator.